contracts/rate_oracles/BaseRateOracle.sol

```diff
@@ -54,10 +56,35 @@ abstract contract BaseRateOracle is IRateOracle, Ownable {
     }
 
     constructor(IERC20Minimal _underlying) {
-        require(address(_underlying) != address(0), "underlying must exist");
         underlying = _underlying;
     }
 
+    /// @dev this must be called at the *end* of the constructor, after the contract member variables have been set, because it needs to read rates.
+    function _populateInitialObservations(
```

Comment:
Is there any possibility to fail populating the rate oracle due to out-of-gas? (if we find that we need to launch a new rate oracle at some later point when we need to feed plenty of data points). I'm thinking if it'd be better to add support for populating the rate in batches.

Reply:
Should be able to add quite a lot of data points so I'd suggest we add that if/when we ever need it.